File: Arcade/Intro/Java/BoxBlur.java
import java.util.Arrays;

final class BoxBlur {

	int[][] boxBlur(int[][] image) {
	    int h = image.length - 2;
	    int w = image[0].length - 2;
	    int[][] sum = new int[h][w];
	    
	    for(int i = 0; i < h; i++) {
	        for(int j = 0; j < w; j++) {
	            for(int y = i; y < i + 3; y++)
	                for(int x = j; x < j + 3; x++)
	                    sum[i][j] += image[y][x];
	            
	            sum[i][j] /= 9;
	        }
	    }
	    
	    return sum;
	}
}

File: Arcade/Intro/javascript/boxBlur.js
function boxBlur(image) {
    let h = image.length - 2, w = image[0].length - 2;
    let sumH = [];
    for(let i = 1; i <= h; i++) {
        let sumW = [];
        for(let j = 1; j <= w; j++) {
            let c =
                image[i - 1][j - 1] + image[i - 1][j] + image[i - 1][j + 1]+
                image[i][j - 1] + image[i][j] + image[i][j + 1]+
                image[i + 1][j - 1] + image[i + 1][j] + image[i + 1][j + 1];
            sumW.push(Math.floor(c / 9));
        }
        sumH.push(sumW);
    }
    
    return sumH;
}